testing/acceptance_test.py

```diff
@@ -1288,3 +1290,62 @@ def test_no_brokenpipeerror_message(pytester: Pytester) -> None:
     ret = popen.wait()
     assert popen.stderr.read() == b""
     assert ret == 1
+
+
+def test_multiple_durations_pluralized_tr_summary(pytester: Pytester) -> None:
+    pytester.makepyfile(
+        """
+        import pytest
+
+        def test_delays():
+            assert True
+        """
+    )
+    result = pytester.runpytest("--durations=5", "--durations-min=10")
+    assert result.ret == ExitCode.OK
+    result.stdout.fnmatch_lines(
+        ["(3 durations slower than 10.000s hidden.  Use -vv to show these durations.)"]
+    )
+
+
+def test_single_duration_pluralized_tr_summary(pytester: Pytester) -> None:
+    pytester.makepyfile(
+        """
+        import pytest
+        import time
+
+        def _sleep(delay):
+            time.sleep(delay)
+
+        def test_one():
+            _sleep(0.05)
+        def test_two():
+            _sleep(0.25)
+        """
+    )
+    result = pytester.runpytest("--durations=2", "--durations-min=0.1")
+    assert result.ret == ExitCode.OK
+    result.stdout.fnmatch_lines(
+        ["(1 duration slower than 0.100s hidden.  Use -vv to show this duration.)"]
+    )
+
+
+def test_durations_zero_lists_all(pytester: Pytester) -> None:
+    pytester.makepyfile(
+        """
+        import pytest
+        import time
+        @pytest.mark.parametrize('_', range(6))
+        def test_this(_):
+            time.sleep(0.1)
+        """
+    )
+    result = pytester.runpytest("--durations=0")
+    assert result.ret == ExitCode.OK
+    # Check 6x setup & 6x teardown are hidden, but 6x call are shown
+    insertable_line = "*test_durations_zero_lists_all.py::test_this*{}*"
+    lines = [insertable_line.format(x) for x in range(6)]
```

Comment:
In my experience tests that depend on a tight sleep like this one tend to be flaky; sometimes an overloaded CI machine will take a bit longer to run a setup function (CI machines tend to have shared CPUs) and then the test will fail.

Don't have a specific suggestion to improve this right now, I suggest we keep this in mind and wait and see.

Reply:
Yeah I was kind of confused in how best to test it (reliably), I can increase the sleep in the test(s) themselves but at the price of 'dead time' in the test run, that said I wouldn't want to introduce something very flaky as its a massive time headache later, open to suggestions

Reply:
One way would be to make unit-tests instead of integration tests; that would require to refactor `pytest_terminal_summary` to extract the relevant logic about building up the list report with pre-defined durations which don't depend on sleep, and then test that.

Reply:
will give it a go, any advice on how to proceed? refactoring a hook seems like the knock on effects could be collossal

Reply:
> will give it a go, any advice on how to proceed? refactoring a hook seems like the knock on effects could be collossal

I meant to just refactor the `pytest_terminal_hook` _implementation_ in `terminal.py`, not to change the `pytest_terminal_hook` spec itself, so the changes should be local and not affect anything else other than `terminal.py`.